crypto/evp/evp_enc.c

```diff
@@ -1248,8 +1252,11 @@ int EVP_CIPHER_get_params(EVP_CIPHER *cipher, OSSL_PARAM params[])
 
 int EVP_CIPHER_CTX_set_params(EVP_CIPHER_CTX *ctx, const OSSL_PARAM params[])
 {
-    if (ctx->cipher != NULL && ctx->cipher->set_ctx_params != NULL)
+    if (ctx->cipher != NULL && ctx->cipher->set_ctx_params != NULL) {
+        /* Assume the worst and invalidate cached values */
+        ctx->key_length = ctx->iv_length = -1;
```

Comment:
Is this assuming set_params is only called once?

Reply:
No.  Each set params call could change the key or IV length so it's cleared every time.